src/Parsing/Impl/Definition/IScopeNode.cs

```diff
@@ -0,0 +1,43 @@
+using Microsoft.Python.Parsing.Ast;
+
+namespace Microsoft.Python.Parsing.Definition {
+    /// <summary>
+    /// Represents a <see cref="Node"/> with a Scope
+    /// </summary>
+    public interface IScopeNode : INode {
+        /// <summary>
+        /// Name of the scope
+        /// </summary>
+        string ScopeName { get; }
+
+        /// <summary>
+        /// Parent of the current scope
+        /// </summary>
+        IScopeNode Parent { get; set; }
+
+        /// <summary>
+        /// Gloabl scope
+        /// </summary>
+        PythonAst GlobalParent { get; }
+
+        /// <summary>
+        /// Holds information about the scope gathered from Parser, mainly used in  <see cref="PythonNameBinder"/>
+        /// </summary>
+        ScopeInfo ScopeInfo { get; }
```

Comment:
This is a concrete type in an interface definition, which I'm not entirely sure about, though it is abstract. @MikhailArkhipov any thoughts on this?

Reply:
I am not sure about introduction of interfaces in the parser. It is all classes so it either needs to be done all interfaces or keep it with classes, otherwise it looks odd mix and match. Is there something that is `IScopeStatement` but not derived from `ScopeStatement`?

Reply:
Ditto for `IScopeInfo` vs `ScopeInfo`. Looks like we are exposing `ScopeInfo` anyway so why the interface then.

Reply:
I asked about this earlier, this was the response: https://github.com/microsoft/python-language-server/pull/1578#issuecomment-535150477

Reply:
I'm personally not against interfacing some things. The problem is that comprehensions are expressions and don't share the same class tree as the existing scope statements, so something is going to have to be done...

Reply:
I think we will have to interface this. The question is should we interface everything to be consistent or leave it partially done. I don’t particularly see a problem with having mix and match but if it’s a public API that might be a problem

Reply:
Yeah, `IScopeNode` may be inevitable, but "ScopeNode require a Statement object as a body" maybe less intrusive, may be just en empty body?

Reply:
Ok I can have null body meaning empty

Reply:
Yeah, `IScopeNode` may be inevitable I thinking other cases like INode etc